src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests.cs

```diff
@@ -7407,6 +7407,611 @@ static string M(object o)
     </method>
   </methods>
 </symbols>
+");
+        }
+
+        [Fact, WorkItem(59050, "https://github.com/dotnet/roslyn/issues/59050")]
+        public void IsPatternInExceptionFilterInAsyncMethod_Spilled()
+        {
+            var source = @"
+using System;
+using System.Threading.Tasks;
+
+static class C
+{
+    static async Task Main()
+    {
+        System.Console.Write(await ExceptionFilterBroken());
+    }
+
+    public static async Task<bool> ExceptionFilterBroken()
+    {
+        try
+        {
+            await ThrowException();
+            return true;
+        }
+        catch (Exception ex) when (ex.InnerException is { Message: ""bad dog"" or ""dog bad"" })
+        {
+            return await TrueAsync();
+        }
+        catch
+        {
+            return false;
+        }
+    }
+
+    static Task ThrowException() => throw new Exception("""", new Exception(""bad dog""));
+    static Task<bool> TrueAsync() => Task.FromResult(true);
+}
```

Comment:
Are we testing cases where a non-matching exception is thrown or no exception is thrown to ensure we're only executing the `catch` block with a filter when appropriate?